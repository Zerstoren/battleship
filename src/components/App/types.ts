import { ILobbyStore, IMainStore } from "../../stores/types";

export enum GameStatus {
  MAIN = 'main',
  LOBBY_LIST = 'list',
  CREATE_LOBBY = 'create',
  WAIT_CONNECT = 'wait',
  SET_SHIPS = 'set_ships',
  GAME = 'game_process',
  GAMEOVER = 'game_end',
}

export interface IGameState {
  gameStatus: GameStatus,
  gameWithUser: boolean,

  setGameStatus: (status: GameStatus) => void,
}

export interface IAppProps {
  mainStore?: IMainStore
}

export interface IProps {
  handleChangeGameStatus: (startGame: GameStatus) => void,
}

export interface ISetLobbyProps {
  handleSetLobby: (startGame: GameStatus, lobby: ILobbyStore) => void
}