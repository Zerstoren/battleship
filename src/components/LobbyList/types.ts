import { ISnapshotOutLobbyStore } from "../../stores/types";

export interface IDataFromServer {
  [key: string]: ISnapshotOutLobbyStore
}
