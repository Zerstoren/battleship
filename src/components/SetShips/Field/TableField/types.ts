import { DragObjectWithType } from "react-dnd/lib/interfaces/hooksApi";

export enum MatrixFill {
  EMPTY = '',
  SHADOW = 's',
  ERR_SHADOW = 'e',
  SET = 'w',
  SET_KILL = 'k',
  MISS = 'm',
}

export interface IMatrix extends Array<Array<MatrixFill>> {}

export interface DragObjectItem extends DragObjectWithType {
  size: number
}