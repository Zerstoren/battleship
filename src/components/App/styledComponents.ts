import styled from "styled-components";

export const CenteredDiv = styled.div`
  width: 600px;
  margin: 0 auto;
  text-align: center;
`;