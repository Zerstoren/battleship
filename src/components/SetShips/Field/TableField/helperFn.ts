import { IMatrix, MatrixFill } from "./types";

export const matrix = (x: number, y: number) : IMatrix => {
  const matrix: Array<Array<MatrixFill>> = [];
  
  for (let dy = 0; dy < y; dy++) {
    let matrixRow: Array<MatrixFill> = [];
    for (let dx = 0; dx < x; dx++) {
      matrixRow.push(MatrixFill.EMPTY);
    }

    matrix.push(matrixRow);
  }

  return matrix;
}

export const matrixClearShadows = (matrix: IMatrix) : IMatrix =>
  matrix.map((y) => y.map((x) => x === MatrixFill.SHADOW ? MatrixFill.EMPTY : x));

export const matrixSetShadow = (matrix: IMatrix, x: number, y: number, shipSize: number) : IMatrix => {
  let dy = -1;

  return matrix.map(yLine => {
    dy += 1;
    let dx = -1;

    return yLine.map(xLine => {
      dx += 1;

      if ((dx >= x && dx <= x + shipSize - 1) && dy === y) {
        return MatrixFill.SHADOW;
      }

      return xLine;
    });
  });
}
