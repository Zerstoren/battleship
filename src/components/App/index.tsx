import { inject, observer } from 'mobx-react';
import React, { FC } from 'react';
import { ILobbyStore } from '../../stores/lobby';
import CreateLobby from '../CreateLobby';
import LobbyList from '../LobbyList';
import Main from '../MainPage';
import SetShips from '../SetShips';
import WaitConnect from '../WaitConnect';
import { AlertError, CenteredDiv } from './styledComponents';
import { GameStatus, IAppProps } from './types';

const App: FC<IAppProps> = inject("mainStore")(observer((props) => {
  const state = props.mainStore;

  const handleGameStart = (startGame: GameStatus) => state!.setGameStatus(startGame);
  const handleSetLobby = (startGame: GameStatus, lobby: ILobbyStore) => state!.setLobby(startGame, lobby);
  const handleWaitLobby = (startGame: GameStatus, lobby: ILobbyStore | null = null) => state!.setLobby(startGame, lobby);
  const handleAcceptError = () => state?.setError('');

  let component = null;
  
  switch(state!.status) {
    case GameStatus.MAIN:
      component = (<Main handleChangeGameStatus={handleGameStart} />);
      break;
    
    case GameStatus.CREATE_LOBBY:
      component = (<CreateLobby handleChangeGameStatus={handleWaitLobby} />);
      break;

    case GameStatus.WAIT_CONNECT:
      component = (<WaitConnect handleChangeGameStatus={handleGameStart} />);
      break;

    case GameStatus.LOBBY_LIST:
      component = (<LobbyList handleSetLobby={handleSetLobby} />);
      break;

    case GameStatus.SET_SHIPS:
      component = (<SetShips handleChangeGameStatus={handleGameStart} />);
  }

  let err = null;
  
  if (state!.error) {
    err = <AlertError>
      <strong>{state!.error}</strong>
      <button type="button" className="btn-close" onClick={handleAcceptError}></button>
    </AlertError>;
  }

  return (
    <>
      {err}
      <CenteredDiv>
        {component}
      </CenteredDiv>
    </>
  );
}));

export default App;
