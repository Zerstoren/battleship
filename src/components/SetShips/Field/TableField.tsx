import React, { FC } from 'react';
import { ILobbyStore } from '../../../stores/lobby';
import { FieldTable } from '../styledComponents';

interface IProp {
  lobby: ILobbyStore
}

const letters: string[] = [
  'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 
  'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 
  'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Z'
];

const TableField: FC<IProp> = (props) => {
  const lobby = props.lobby;

  let tr: JSX.Element[] = [];
  let theadTd: JSX.Element[] = [<td key="empty"></td>];

  for (let y = 0; y < lobby.y; y++) {
    let td: JSX.Element[] = [<td key={`head-${y}`}>{y+1}</td>];

    for (let x = 0; x < lobby.x; x++) {
      td.push(<td key={`${y}-${x}`}></td>);
    }

    tr.push(<tr key={y}>{td}</tr>);
  }

  for (let x = 0; x < lobby.x; x++) {
    theadTd.push(<td key={`head-${x}`}>{letters[x]}</td>);
  }
  
  return (
    <>
      <FieldTable sizeX={lobby.x} sizeY={lobby.y}>
        <thead>
          <tr>
            {theadTd}
          </tr>
        </thead>
        <tbody>
          {tr}
        </tbody>
      </FieldTable>
    </>
  );
};

export default TableField;
