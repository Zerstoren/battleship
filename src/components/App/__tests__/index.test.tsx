import { Provider } from 'mobx-react';
import React from 'react';
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import renderer from 'react-test-renderer';
import App from '..';
import { MainStore } from '../../../stores/mainStore';
import { matrix } from "../../SetShips/Field/TableField/helperFn";
import { GameStatus } from '../types';

describe('App snapshots', () => {
  it('Check main page', () => {
    let mainStore = MainStore.create({
      status: GameStatus.MAIN
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check create lobby', () => {
    let mainStore = MainStore.create({
      status: GameStatus.CREATE_LOBBY
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check lobby list', () => {
    let mainStore = MainStore.create({
      status: GameStatus.LOBBY_LIST
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check wait connect', () => {
    let mainStore = MainStore.create({
      status: GameStatus.WAIT_CONNECT
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check set ships', () => {
    let mainStore = MainStore.create({
      status: GameStatus.SET_SHIPS,
      currentLobby: {
        lobbyName: 'Ok',
        x: 10,
        y: 10,
        ships4n: 1,
        ships3n: 1,
        ships2n: 1,
        ships1n: 1,
      }
    });
    
    const wrapper = renderer.create(
      <Provider mainStore={mainStore}>
        <DndProvider backend={HTML5Backend}>
          <App />
        </DndProvider>
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check game', () => {
    let mainStore = MainStore.create({
      status: GameStatus.GAME,
      gameMatrix: matrix(10, 10),
      currentLobby: {
        lobbyName: 'Ok',
        x: 10,
        y: 10,
        ships4n: 1,
        ships3n: 1,
        ships2n: 1,
        ships1n: 1,
      }
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check game win', () => {
    let mainStore = MainStore.create({
      status: GameStatus.GAMEWIN
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Check game over', () => {
    let mainStore = MainStore.create({
      status: GameStatus.GAMEOVER
    });
    
    const wrapper = renderer.create(<Provider mainStore={mainStore}><App /></Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
})