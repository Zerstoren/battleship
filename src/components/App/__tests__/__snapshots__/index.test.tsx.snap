// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots Check create lobby 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    New Lobby
  </h1>
  <div
    className="sc-gsTCUz hIfhYJ"
  >
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        autoFocus={true}
        className="sc-hKgILt form-control false"
        name="lobbyName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <label>
        Lobby Name
      </label>
    </div>
  </div>
  <div
    className="sc-gsTCUz hIfhYJ d-flex justify-content-between"
  >
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="x"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={10}
      />
      <label>
        Field size: X
      </label>
    </div>
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="y"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={10}
      />
      <label>
        Field size: Y
      </label>
    </div>
  </div>
  <div
    className="sc-gsTCUz hIfhYJ d-flex justify-content-between"
  >
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="ships4n"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={1}
      />
      <label>
        4x ships count
      </label>
    </div>
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="ships3n"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={2}
      />
      <label>
        3x ships count
      </label>
    </div>
  </div>
  <div
    className="sc-gsTCUz hIfhYJ d-flex justify-content-between"
  >
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="ships2n"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={3}
      />
      <label>
        2x ships count
      </label>
    </div>
    <div
      className="sc-dlfnbm form-floating"
    >
      <input
        className="sc-hKgILt form-control undefined"
        name="ships1n"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={4}
      />
      <label>
        1x ships count
      </label>
    </div>
  </div>
  <div
    className="sc-gsTCUz hIfhYJ d-flex justify-content-evenly"
  >
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Create lobby
    </button>
  </div>
</div>
`;

exports[`App snapshots Check game 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    Fight
  </h1>
  <div
    className="sc-jSgupP d-flex"
  >
    <table
      className="sc-gKsewC sc-iBPRYJ hPxzGb twbZX table table-bordered"
    >
      <thead>
        <tr>
          <td />
          <td>
            A
          </td>
          <td>
            B
          </td>
          <td>
            C
          </td>
          <td>
            D
          </td>
          <td>
            E
          </td>
          <td>
            F
          </td>
          <td>
            G
          </td>
          <td>
            H
          </td>
          <td>
            I
          </td>
          <td>
            J
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            2
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            3
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            4
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            5
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            6
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            7
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            8
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            9
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
        <tr>
          <td>
            10
          </td>
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
          <td
            className=""
            onClick={[Function]}
          />
        </tr>
      </tbody>
    </table>
    <table
      className="sc-gKsewC hPxzGb table table-bordered"
    >
      <thead>
        <tr>
          <td />
          <td>
            A
          </td>
          <td>
            B
          </td>
          <td>
            C
          </td>
          <td>
            D
          </td>
          <td>
            E
          </td>
          <td>
            F
          </td>
          <td>
            G
          </td>
          <td>
            H
          </td>
          <td>
            I
          </td>
          <td>
            J
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            2
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            3
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            4
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            5
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            6
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            7
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            8
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            9
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
        <tr>
          <td>
            10
          </td>
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
          <td
            className=""
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App snapshots Check game over 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    You are loose
  </h1>
  <button
    className="sc-eCssSg btn btn-primary"
    onClick={[Function]}
  >
    To main menu
  </button>
</div>
`;

exports[`App snapshots Check game win 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    You are win
  </h1>
  <button
    className="sc-eCssSg btn btn-primary"
    onClick={[Function]}
  >
    To main menu
  </button>
</div>
`;

exports[`App snapshots Check lobby list 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    Lobby list
  </h1>
  <table
    className="sc-fubCfw table table-striped table-hover"
  >
    <thead>
      <tr>
        <td />
        <td>
          Lobby name
        </td>
        <td>
          Field size
        </td>
        <td>
          Ships 4x/3x/2x/1x
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`App snapshots Check main page 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    Battleship
  </h1>
  <div>
    <button
      className="sc-pFZIQ fwORzk btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Search opponent
    </button>
    <button
      className="sc-pFZIQ sc-jrAGrp fwORzk btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Create lobby
    </button>
  </div>
</div>
`;

exports[`App snapshots Check set ships 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    Set ships
  </h1>
  <div
    className="d-flex"
  >
    <div>
      <table
        className="sc-kEjbxe jokKlb table table-bordered"
      >
        <thead>
          <tr>
            <td />
            <td>
              A
            </td>
            <td>
              B
            </td>
            <td>
              C
            </td>
            <td>
              D
            </td>
            <td>
              E
            </td>
            <td>
              F
            </td>
            <td>
              G
            </td>
            <td>
              H
            </td>
            <td>
              I
            </td>
            <td>
              J
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              2
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              3
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              4
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              5
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              6
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              7
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              8
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              9
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
          <tr>
            <td>
              10
            </td>
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
            <td
              className=""
              onClick={[Function]}
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="sc-iqHYGH erCuwz"
    >
      <div
        className="sc-crrsfI hCyPpV d-flex"
      >
        <div
          className="sc-hBEYos gFwwHG d-flex undefined"
        >
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
        </div>
        <span>
           
          x
          1
        </span>
      </div>
      <div
        className="sc-crrsfI hCyPpV d-flex"
      >
        <div
          className="sc-hBEYos gFwwHG d-flex undefined"
        >
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
        </div>
        <span>
           
          x
          1
        </span>
      </div>
      <div
        className="sc-crrsfI hCyPpV d-flex"
      >
        <div
          className="sc-hBEYos gFwwHG d-flex undefined"
        >
          <div
            className="sc-kstrdz cLMXsX"
          />
          <div
            className="sc-kstrdz cLMXsX"
          />
        </div>
        <span>
           
          x
          1
        </span>
      </div>
      <div
        className="sc-crrsfI hCyPpV d-flex"
      >
        <div
          className="sc-hBEYos gFwwHG d-flex undefined"
        >
          <div
            className="sc-kstrdz cLMXsX"
          />
        </div>
        <span>
           
          x
          1
        </span>
      </div>
      <div
        className="sc-dQppl dijAIy"
      />
    </div>
  </div>
</div>
`;

exports[`App snapshots Check wait connect 1`] = `
<div
  className="sc-fodVxV gZZuTH"
>
  <h1
    className="sc-bdfBwQ display-3"
  >
    Wait connect
  </h1>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Discard
  </button>
</div>
`;
